hmtp.widget.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+add_action( 'widgets_init', function() { register_widget( 'HMTP_Widget' ); });
+
+class HMTP_Widget extends WP_Widget {
+
+	public function __construct() {
+		parent::__construct(
+			'HMTP_Widget', // Base ID
+			'Top Post Widget', // Name
+			array( 'description' => 'Display top posts using Google Analytics data.' ) // Args
+		);
+	}
+
+	/**
+	 * Front-end display of widget.
+	 *
+	 * @see WP_Widget::widget()
+	 *
+	 * @param array $args     Widget arguments.
+	 * @param array $instance Saved values from database.
+	 */
+	public function widget( $args, $instance ) {
+
+		// Demo.
+		$results = hmtp_get_top_posts( $instance['args'] );
+
+		if ( ! $results )
+			return;
+
+		echo $args['before_widget'];
+
+		if ( $instance['title'] ) {
+			echo $args['before_title'];
+			echo esc_html( $instance['title'] );
+			echo $args['after_title'];
+		}
+
+		ob_start();
+
+		?>
+
+		<ol class="hmtp-widget">
+			<?php foreach ( $results as $post ) : ?>
+				<li><?php printf(
+					'<a href="%s">%s</a>',
+					esc_url( get_permalink( $post['post_id'] ) ),
+					esc_html( get_the_title( $post['post_id'] ) )
+				); ?></li>
+			<?php endforeach; ?>
+		</ol>
+
+		<?php
+
+		echo apply_filters( 'hmtp-widget-output', ob_get_clean(), $results );
+
+		echo $args['after_widget'];
+
+	}
+
+
+	/**
+	 * Back-end widget form.
+	 *
+	 * @see WP_Widget::form()
+	 *
+	 * @param array $instance Previously saved values from database.
+	 */
+ 	public function form( $instance ) {
+
+ 		$title = isset( $instance[ 'title' ] ) ? $instance[ 'title' ] : 'Most Popular';
+
+ 		$args = wp_parse_args(
+ 			(array) $instance['args'],
```

Comment:
There is no check ensuring that this key exists and it is thusly throwing an notice:

```
[30-Oct-2018 18:45:00 UTC] PHP Notice:  Undefined index: args in /chassis/content/plugins/hm-top-posts/hmtp.widget.php on line 74
[30-Oct-2018 18:45:00 UTC] PHP Stack trace:
[30-Oct-2018 18:45:00 UTC] PHP   1. {main}() /chassis/wordpress/wp-admin/widgets.php:0
[30-Oct-2018 18:45:00 UTC] PHP   2. wp_list_widgets() /chassis/wordpress/wp-admin/widgets.php:401
[30-Oct-2018 18:45:00 UTC] PHP   3. wp_widget_control() /chassis/wordpress/wp-admin/includes/widgets.php:48
[30-Oct-2018 18:45:00 UTC] PHP   4. HMTP_Widget->form_callback() /chassis/wordpress/wp-admin/includes/widgets.php:243
[30-Oct-2018 18:45:00 UTC] PHP   5. HMTP_Widget->form() /chassis/wordpress/wp-includes/class-wp-widget.php:507
[30-Oct-2018 18:45:17 UTC] PHP Notice:  Undefined index: args in /chassis/content/plugins/hm-top-posts/hmtp.widget.php on line 74
[30-Oct-2018 18:45:17 UTC] PHP Stack trace:
[30-Oct-2018 18:45:17 UTC] PHP   1. {main}() /chassis/wordpress/wp-admin/widgets.php:0
[30-Oct-2018 18:45:17 UTC] PHP   2. wp_list_widgets() /chassis/wordpress/wp-admin/widgets.php:401
[30-Oct-2018 18:45:17 UTC] PHP   3. wp_widget_control() /chassis/wordpress/wp-admin/includes/widgets.php:48
[30-Oct-2018 18:45:17 UTC] PHP   4. HMTP_Widget->form_callback() /chassis/wordpress/wp-admin/includes/widgets.php:243
[30-Oct-2018 18:45:17 UTC] PHP   5. HMTP_Widget->form() /chassis/wordpress/wp-includes/class-wp-widget.php:507
```